fluent-package/templates/package-scripts/fluent-package/deb/postinst

```diff
@@ -22,38 +20,113 @@ add_system_user() {
+	fi
+    fi
 }
 
 fixperms() {
-    # If statoverride entry exits, use --force-all to ensure statoverride-add will work.
-    # If statoverride entry doen't exit, no need to --force-all.
+    # If statoverride entry doesn't exit, set it.
     dpkg-statoverride --list /var/run/<%= package_dir %> >/dev/null || \
         dpkg-statoverride --update --add _<%= service_name %> _<%= service_name %> 0755 /var/run/<%= package_dir %>
-    dpkg-statoverride --list /var/run/<%= package_dir %> >/dev/null && \
-        dpkg-statoverride --force-all --update --add _<%= service_name %> _<%= service_name %> 0755 /var/run/<%= package_dir %>
-    dpkg-statoverride --list /etc/<%= compat_package_dir %> >/dev/null || \
-        dpkg-statoverride --update --add _<%= service_name %> _<%= service_name %> 0755 /etc/<%= compat_package_dir %>
+    dpkg-statoverride --list /etc/<%= package_dir %> >/dev/null || \
+        dpkg-statoverride --update --add _<%= service_name %> _<%= service_name %> 0755 /etc/<%= package_dir %>
+    dpkg-statoverride --list /var/log/<%= package_dir %> >/dev/null || \
+        dpkg-statoverride --update --add _<%= service_name %> _<%= service_name %> 0755 /var/log/<%= package_dir %>
+    # Remove obsolete statoverride
+    dpkg-statoverride --list /var/run/<%= compat_package_dir %> >/dev/null && \
+	dpkg-statoverride --remove /var/run/<%= compat_package_dir %>
     dpkg-statoverride --list /etc/<%= compat_package_dir %> >/dev/null && \
-        dpkg-statoverride --force-all --update --add _<%= service_name %> _<%= service_name %> 0755 /etc/<%= compat_package_dir %>
-    dpkg-statoverride --list /var/log/<%= compat_package_dir %> >/dev/null || \
-        dpkg-statoverride --update --add _<%= service_name %> _<%= service_name %> 0755 /var/log/<%= compat_package_dir %>
+	dpkg-statoverride --remove /etc/<%= compat_package_dir %>
     dpkg-statoverride --list /var/log/<%= compat_package_dir %> >/dev/null && \
-        dpkg-statoverride --force-all --update --add _<%= service_name %> _<%= service_name %> 0755 /var/log/<%= compat_package_dir %>
-    # Cleanup /var/run/<%= compat_package_dir %>
-    dpkg-statoverride --force-all --remove /var/run/<%= compat_package_dir %>
+	dpkg-statoverride --remove /var/log/<%= compat_package_dir %>
+}
+
+fixconditions() {
+    if [ -f $v4migration ]; then
+	if [ -d /etc/<%= compat_package_dir %> ]; then
+	    rm -fr /etc/<%= compat_package_dir %>
+	    ln -sf /etc/<%= package_dir %> /etc/<%= compat_package_dir %>
+	fi
+	if [ -d /var/log/<%= compat_package_dir %> ]; then
+	    rm -fr /var/log/<%= compat_package_dir %>
+	    ln -sf /var/log/<%= package_dir %> /var/log/<%= compat_package_dir %>
+	fi
+	rm -f $v4migration
+    fi
+    if [ -f $v4migration_with_restart ]; then
+	systemctl restart <%= service_name %>
+	rm -f $v4migration_with_restart
+    fi
 }
 
+
 case "$1" in
     configure)
         add_system_user
         add_directories
         fixperms
-        ;;
+        fixconditions
+	;;
```

Comment:
If there is no problem, it may be more maintainable to separate the migration process from the normal process.
I don't know what to do with rpm `.spec` file, though,,,

I understand that there are things that should be prioritized over maintainability, especially for now.
It is just an idea.

```sh
...

migrate_from_v4_main_process() {
    if [ -d /etc/<%= compat_package_dir %> -a ! -h /etc/<%= compat_package_dir %> ]; then
        ...
    fi
    ...
}

migrate_from_v4_post_process() {
    # Remove obsolete statoverride
    ...

    # Fix conditions
    ...

    if [ -f $v4migration_with_restart ]; then
        systemctl restart <%= service_name %>
        rm -f $v4migration_with_restart
    fi

    rm -f $v4migration
}

if [ $prevver ... ] then;
    touch $v4migration
fi

case "$1" in
    configure)
        add_system_user
        add_directories
        if [ -f $v4migration ] then;
            migrate_from_v4_main_process
        fi
        fixperms
        if [ -f $v4migration ] then;
            migrate_from_v4_post_process
        fi
        ;;
...
```

Reply:
extracting migration procedure is a good approarch, I'll fix it.